generated/nidmm/tests/test_session.py

```diff
@@ -75,18 +88,29 @@ def test_multiple_return_params(self):
             assert(hour == 10)
             assert(minute == 12)
             self.patched_ctypes_library.niDMM_GetCalDateAndTime.assert_called_once_with(SESSION_NUM_FOR_TEST, 0, ANY, ANY, ANY, ANY, ANY)
-            assert self.patched_errors._handle_error.call_count == 2
-            self.patched_errors._handle_error.assert_called_with(session, self.patched_ctypes_library.niDMM_GetCalDateAndTime.return_value)
 
     # Test getting a string attribute (IVI dance to get string)
     def test_get_string_attribute(self):
         self.patched_ctypes_library.niDMM_GetAttributeViString.side_effect = self.side_effects_helper.niDMM_GetAttributeViString
         self.side_effects_helper['GetAttributeViString']['attributeValue'] = 'Testing is fun?'
         with nidmm.Session('dev1') as session:
             attr_string = session._get_attribute_vi_string("", 5)
-            assert(attr_string == 'Testing is fun?')
-            assert self.patched_errors._handle_error.call_count == 2
-            assert self.patched_ctypes_library.niDMM_GetAttributeViString.call_count == 2
+            assert attr_string == 'Testing is fun?'
+
+    def test_get_string_attribute_with_error(self):
+        self.patched_ctypes_library.niDMM_GetAttributeViString.side_effect = self.side_effects_helper.niDMM_GetAttributeViString
+        self.side_effects_helper['GetAttributeViString']['attributeValue'] = 'Testing is fun?'
+        self.side_effects_helper['GetAttributeViString']['return'] = -1234
+        self.patched_ctypes_library.niDMM_GetError.side_effect = self.side_effects_helper.niDMM_GetError
+        self.side_effects_helper['GetError']['errorCode'] = -1234
+        self.side_effects_helper['GetError']['description'] = "ascending order"
+        with nidmm.Session('dev1') as session:
+            try:
+                session._get_attribute_vi_string("", 5)
```

Comment:
I don't think we need to have unit tests for private methods (although I'm not sure about that).
We should test this using the public API.